subsys/bootloader/cmake/sign.cmake

```diff
@@ -67,32 +81,71 @@ endif()
 
 add_custom_command(
   OUTPUT
-  ${signature_file}
+  ${HASH_FILE}
   COMMAND
-  ${hashcmd}
+  ${HASH_CMD}
+  DEPENDS
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  WORKING_DIRECTORY
+  ${PROJECT_BINARY_DIR}
+  COMMENT
+  "Creating hash of application"
+  USES_TERMINAL
+  )
+
+add_custom_target(
+  hash_file_target
```

Comment:
In general, the pattern used to get a target for a file, which can be depended on is as follows:
```
add_custom_command(
OUTPUT my_file
COMMAND
  make my file
DEPENDS
  many
  different
  targets
  which
  we
  depend
  on
)

add_custom_target(my_file_target DEPENDS my_file)
```

So all the dependencies for creating the file is listed in the add_custom_command, while the only dependency listed in the custom target created is the output file itself.

This comment applies to this whole file.

Reply:
Think I've addressed this now